cpp/src/arrow/type.h

```diff
@@ -174,6 +175,25 @@ class ARROW_EXPORT DataType : public detail::Fingerprintable {
   /// \brief Return the type category of the storage type
   virtual Type::type storage_id() const { return id_; }
 
+  /// \brief Returns the type's fixed byte width, if any. Returns -1
+  /// for non-fixed-width types, and should only be used for
+  /// subclasses of FixedWidthType
+  virtual int32_t byte_width() const {
```

Comment:
I added this because getting the byte width from a fixed width type is quite tedious at the moment (there's many places where we do the cast-- I think having the bit_width()/byte_width() methods available on non-fixed-width types (returning -1) is pretty harmless, but if others don't like it we can remove these and add a helper function or something)

Reply:
This method doesn't need to be virtual, does it?